api/repositories/service_instance_repository.go

```diff
@@ -456,6 +456,36 @@ func (r *ServiceInstanceRepo) GetServiceInstance(ctx context.Context, authInfo a
 	return cfServiceInstanceToRecord(*serviceInstance), nil
 }
 
+func (r *ServiceInstanceRepo) GetServiceInstanceCredentials(ctx context.Context, authInfo authorization.Info, secretName string) (map[string]any, error) {
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return map[string]any{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	namespace, err := r.namespaceRetriever.NamespaceFor(ctx, secretName, ServiceInstanceResourceType)
+	if err != nil {
```

Comment:
Maybe this is not the best way to do this, because we get the namespace from the service instance. Should we introduce the corev1.Secret Resource here: https://github.com/cloudfoundry/korifi/blob/main/api/repositories/namespace_retriever.go ? This seams the most reasonable approach to me.

Reply:
If we make the method to receive the instance guid (rather than the secret name), we could get the namespace for the service instance (something I believe we already have), not for the secret.